contracts/SimpleRegistry.sol

```diff
@@ -53,7 +53,10 @@ contract SimpleRegistry is Owned, MetadataRegistry, OwnerRegistry, ReverseRegist
 	}
 
 	modifier whenEntry(string _name) {
-		require(!entries[keccak256(bytes(_name))].deleted);
+		require(
+			!entries[keccak256(bytes(_name))].deleted &&
+			entries[keccak256(bytes(_name))].owner != address(0)
```

Comment:
This additional address check should also be added to the `whenEntryRaw` modifier.

Reply:
I can add it, but it only works if I take out the `whenEntryRaw` modifier from the `reserve()` function. Otherwise it is impossible to reserve new names.

Reply:
pushed some modifications @gautamdhameja